### ISSUE TITLE
Set Torrent Download Folder on Supported Clients (transmission deluge etc)

### ISSUE THREAD

**Issue by xInsertx** [Sonarr/Sonarr#744]
Can we please have the ability to set the download path on clients that support it. I know this was mentioned in #587 but using category to set a subfolder does not offer much flexibility. This is very limiting

Pretty much the only reason I personally (aswell as most of my friends) have not moved from sickrage.

Thanks for you time.

**Comment by Taloth** [comment]
Your mentioned Deluge, did you know Deluge allows you to specify the destination folder per label/category?

**Comment by xInsertx** [comment]
@Taloth Thanks for the reply. Yes, but i mainly use transmission. I simply referenced deluge as another example supporting destination folder in the API.

**Comment by xelra** [comment]
@xInsertx The download folder for transmission is the category you set. So if you set a category `sonarr`, the folder will be `/downloadRoot/.sonarr`.

**Comment by Taloth** [comment]
@xelra yes, but he wants to set it to an arbitrary directory, and still have Sonarr detect the category.
Also the dot is no longer added to the category.

**Comment by xInsertx** [comment]
@Taloth that is correct. if i cannot set /path/to/download than its essentially useless. Sickbeard/Sickrage have supported this for quite a while. 

Since your guys added anime support i keep trying to move over, but until you support this i'll need to continue to use sickrage.

**Comment by Taloth** [comment]
Sorry to ask, but how is this useless? What is so unique about your directory structure that using subdirectories is entirely unacceptable?

**Comment by xInsertx** [comment]
@Taloth Useless might be abit of a overstatement; but it would be difficult to work with (esp with transmission), I could solve the issues by using deluge but this will introduce separate issues.

Without going over complicated w/o other scripts and programs:
-/downloads
---/incomplete
---/complete *this is the default download dir for transmission (where anything not automated goes)
---/seeding
-----/sonarr *this is where i want the files to be downloaded to, which is currently not possible.
-----/sickrage *this works fine with sickbeard/sickrage
-----/many other subfolders such as btsync

**This is without including any external scripts that run, hard links/symlinks or going across volumes.**

Due to the current limitation all sonarr downloads could only be stored in /downloads/complete/<category> at best. We cannot traverse any folder below or beyond.

Another use-case is some people store there seeding directory on another volume so that there main pool can sleep to save energy. This would be impossible unless the default directory was also stored there.

TL:DR there is many use cases for adding this feature. It does introduce complexity if used incorrectly but it adds greater filesystem flexibility.

**Comment by aroundmyroom** [comment]
is it meant that you store only the file once and transmission still can find the file to be seeded?
Indeed right now with Deluge all files will be downloaded as in above example and all the files downloaded 100% are stored in the 'complete' folder than copied by Sonarr to my TV folder

**Comment by xInsertx** [comment]
How my sickrage and other apps work for seeding related stuff. Anything I manually download goes to complete by default. Anything my apps download that require seeding are downloaded to there appropriate seeding\appname folder. From here the apps can copy/hardlink whilst transmission remains seeding. This keeps the folder structure clean and allows flexibility.

**Comment by LargeTalons** [comment]
How is manually setting a path useless and difficult to work with? Both Couchpotato and Sickrage do it just fine...so why is Sonarr so special that it cannot be done?

**Comment by Taloth** [comment]
We aren't turning our nose up, the issue isn't closed, we just don't make a habit of adding features / config options that don't have a clear usecase.

That's why I asked about his directory structure and xInsertx explained that in detail. I just didn't reply yet.

Essentially this is a shortcoming in Transmission, iirc the Mac version has labels/categories, but the other version still don't.
rTorrent does have labels, but doesn't attached a dedicated directory to it, so basically the same issue.

I talked to the other devs a couple of days ago, we'll add the download dir option.

**Comment by xInsertx** [comment]
@Taloth Thanks for the reply. I'm looking forward to the pull request when this is added.

**Comment by LargeTalons** [comment]
Thank you,

**Comment by bcg62** [comment]
Is this still pending? I could also use this functionality.

**Comment by xInsertx** [comment]
@bcg62 i don't think its even been started (could be wrong, but couldn't
see it in development branch either)

**Comment by deandmx** [comment]
I'd like to throw my hat in for a use case.

I use Deluge to download torrents, Sonarr for TV and CouchPotato for films. I have movies on one external drive and tv shows on another external drive. My internal is a small SSD. I've set the default download folder for CouchPotato to be the external drive where the movies live. I'd like to do the same with Sonarr. Reason being, space is tight and it saves time and disk grinding if the files are downloaded to the drive where they'll end up rather than being downloaded, then copied. Plus Time Machine backs up the internal which adds to the size and time of the backups.

**Comment by LargeTalons** [comment]
I feel like the devs still don’t view this feature as something they need to spend time on. 

 

As such, I’ve gone back to Sickrage. Not to mention the fact that Sickrage does Symbolic links and Sonarr only does Hard Links. When I post processed all the stuff Sonarr had downloaded, I ended up freeing up over 300GB of space on my appdata drive for Unraid. That’s insane. 

 

We shouldn’t have to plead our case for features like this when similar programs have it. I’ll come back when the features are there.

**Comment by xInsertx** [comment]
@largetalons that's pretty much exactly where im at.

I was actually surprised when you said they didn't support symlinks
(actually went back and checked). That is a pretty space saving feature if
you use multiple volumes. Luckily space is nolonger a issue for me (for
now....)

**Comment by coolboiime** [comment]
I seem to want this feature too on Transmission. I didn't know that it was missing until I actually needed it.

**Comment by cgoubert** [comment]
I'd like this feature too. I run all apps (Transmission, Sonarr, Sabnzbd...) in docker containers, and not being able to set the download folder for Transmission is a severe limitation.

**Comment by Krun** [comment]
My use case:

I have a server with transmission, plex and sonarr. I want to have my series managed by sonarr, but also be able to use transmission web ui to download other stuff. This stuff will by default be saved to /my/downloadlocation and sonarr stuff to /my/downloadlocation/sonarr

Then, I would like to have my non-sonar downloads managed by another program (say, a webdav server), to which I would like to give ownership of my transmission's default download folder.

However, since now sonarr's folder is inevitably inside transmission's download folder, I can't do this easily. If I give my web server ownership of transmission's default folder, it will also catch sonarr's.

I would like to make transmission download by default to /web/managed/directory, and sonarr downloads to /sonarr/managed/directory, so I could tell each program to watch their respective folder.

Of course at the moment I can achieve this by running post-download scripts or by setting up a complex permissions structure, but seems logical to me that sonarr would let me specify a download folder in the same way that it lets me set a category for transmission.

**Comment by intelligence** [comment]
It seems I cannot start using Sonarr properly until this feature is implemented. I'm on FreeNAS and both Couchpotato and Sickrage (which I'm currently using) is setup to pull stuff into /Downloads/Movies/ and /Downloads/Series/. Upon completion a hardlink  to /Movies/ or /Series/ is created.

**Comment by xInsertx** [comment]
@Taloth any progress on this issue? I couldn't find any notes/pr relating
to it.

**Comment by mmeidlinger** [comment]
While 050143473aa0e7af93c0890c8e973b4fe055ce26 provides a solution to the  Transmission torrent client, it would be nice to have the same for Deluge. My use case: I too have different types of media on different physical volumes (in my case: zfs data sets). With the current setup and a single Deluge default download directory, all media has to be written twice, once temporary to the download directory and a second time when using`mv` across file systems. Deluge Labels don't really help here, because they only apply the location change once the download is finished. A solution such as 050143473aa0e7af93c0890c8e973b4fe055ce26 for Deluge would drastically reduce the stress on storage hardware.

**Comment by 88fingerslukee** [comment]
I see that this has been changed but I can't find the setting. Shouldn't it be in the download client's settings? What am I missing here?
